test/factorize.jl

```diff
@@ -296,3 +296,88 @@ end
         end
     end
 end
+
+# Test complete BlockLanczos factorization
+@testset "Complete BlockLanczos factorization " for mode in (:vector, :inplace, :outplace)
+    scalartypes = mode === :vector ? (Float32, Float64, ComplexF32, ComplexF64) :
+                  (ComplexF64,)
+    using KrylovKit: EACHITERATION_LEVEL
+    @testset for T in scalartypes
+        A = rand(T, (N, N))
+        A = (A + A') / 2
```

Comment:
Maybe it is useful to have a small utility function that generates some matrix with at least one repeated eigenvalue with a given multiplicity, that can be chosen equal to `block_size`, something like:

```julia
function mat_with_eigrepition(T, N, multiplicity)
    U = qr(randn(T, (N, N))).Q # Haar random matrix
    D = sort(randn(real(T), N))
    i = 0
    while multiplicity >= 2 && (i + multiplicity) <= N
        D[i .+ (1:multiplicity)] .= D[i+1]
        i += multiplicity
        multiplicity -= 1
    end
    A = U * Diagonal(D) * U'
    return (A + A')/2
end
```

which generates a random Hermitian matrix with eigenvalues with multiplicities `multiplicity`, `multiplicity-1`, `multiplicity-2`, ... in its `:SR` eigenvalues.